### PR TITLE
Add --delivery-function option to add the delivery function example to create-contentful-app [MONET-1334]

### PR DESCRIPTION
**Pull request by BasKiers**
The PR includes the following changes
- Adds a new `--delivery-function` option to `create-contentful-app`
- Merges content from `package.json` in delivery function template folder into the project's `package.json`
- Fixes a small bug encountered during testing which prevented the app directory from being removed on an unexpected error

### CODE REVIEW

File: packages/contentful--create-contentful-app/src/includeAppAction.ts
Comment (kdamball) [inline packages/contentful--create-contentful-app/src/includeAppAction.ts:36]: If we're moving to async, we can do this js file and json merge above in parallel :)

File: packages/contentful--create-contentful-app/src/includeDeliveryFunction.ts
Comment (kdamball) [inline packages/contentful--create-contentful-app/src/includeDeliveryFunction.ts:35]: Same thing here

File: packages/contentful--create-contentful-app/src/index.ts
Comment (kdamball) [inline packages/contentful--create-contentful-app/src/index.ts:138]: parallelize here too, with the action script above

File: packages/contentful--create-contentful-app/src/utils/file.ts
Comment (kdamball) [inline packages/contentful--create-contentful-app/src/utils/file.ts:43]: nit: fwiw usually it's recommended to just try writing to the file directly and catch the `FILE_NOT_FOUND` equivalent (https://nodejs.org/api/fs.html#fsexistspath-callback). I guess here we kinda need it for a cleaner error

File: packages/contentful--create-contentful-app/src/index.ts
Comment (BasKiers) [reply packages/contentful--create-contentful-app/src/index.ts:138]: Personally I feel that in this instance the gained performance does not weigh up against the increased code complexity of parallelising these two tasks. As the consumer of the library, since I'm only calling this function once in the entire lifecycle of my application, I'd rather wait a few more seconds and have the logs read nicely step by step.

**Review by kdamball — APPROVED** [review]

**Comment by ghepting** [comment]
Hey @BasKiers! moving forward can you please give @contentful/team-developer-experience a chance to review things before merging? And some additional context/screenshots/examples to help us understand and possibly contribute to the overall DevExp of new features like this would be splendid. We don't want to be forced to configure ourselves as required core reviewers, but we do now own and are responsible for this and other repositories that were previously managed by @contentful/team-extensibility and it's important that we start to work towards better awareness, communication, and coordination on changes that are going to be proposed or take place.

**Pull request merged**
